js_modules/dagster-ui/packages/ui-core/src/assets/useAssetDefinition.ts

```diff
@@ -0,0 +1,51 @@
+import {useContext, useMemo} from 'react';
+
+import {ASSET_VIEW_DEFINITION_QUERY} from './AssetView';
+import {useAllAssets} from './AssetsCatalogTable';
+import {AssetKey} from './types';
+import {AppContext} from '../app/AppContext';
+import {tokenForAssetKey} from '../asset-graph/Utils';
+import {
+  AssetViewDefinitionQuery,
+  AssetViewDefinitionQueryVariables,
+  AssetViewDefinitionQueryVersion,
+} from './types/AssetView.types';
+import {useIndexedDBCachedQuery} from '../search/useIndexedDBCachedQuery';
+
+export const useAssetDefinition = (assetKey: AssetKey) => {
+  const {assets} = useAllAssets();
+  const cachedDefinition = useMemo(
+    () =>
+      assets?.find((asset) => tokenForAssetKey(asset.key) === tokenForAssetKey(assetKey))
+        ?.definition,
+    [assetKey, assets],
+  );
+  const {localCacheIdPrefix} = useContext(AppContext);
+  const result = useIndexedDBCachedQuery<
+    AssetViewDefinitionQuery,
+    AssetViewDefinitionQueryVariables
+  >({
+    query: ASSET_VIEW_DEFINITION_QUERY,
+    variables: {assetKey: {path: assetKey.path}},
+    key: `${localCacheIdPrefix}/asset-definition-${assetKey.path.join(',')}`,
+    version: AssetViewDefinitionQueryVersion,
+  });
+
+  const {assetOrError} = result.data || result.previousData || {};
+  const asset = assetOrError && assetOrError.__typename === 'Asset' ? assetOrError : null;
+  if (!asset) {
+    return {
+      definitionQueryResult: result,
+      definition: null,
+      lastMaterialization: null,
+      cachedDefinition,
```

Comment:
Hmm I wonder if we should bother with this case where we override the cached definition? It might actually be preferable for this page to always use the `useAllAssets()` version of the asset. Since that version is used ~most other places, it seems like it'd just be confusing if the asset page showed you a newer one than the rest of the app?